src/cljs/owlet_ui/events/contentful.cljs

```diff
@@ -70,24 +69,41 @@
             :activity-titles activity-titles)
        :dispatch [route-dispatch route-param]})))
 
-;TODO: add :show-skill, :show-activity
+
+; route dispatches
 
 (rf/reg-event-fx
   :show-branches
   (fn [_ _]
-    {:dispatch [:set-active-view :branches-view]}))
+    {:dispatch-n (list [:set-active-view :branches-view]
+                       [:set-active-document-title! "Branches"])}))
+
+(rf/reg-event-fx
+  :show-branch
+  (fn [_ [_ route-param]]
+    {:dispatch-n (list [:set-active-view :filtered-activities-view "branch"]
+                       [:filter-activities-by-search-term route-param]
+                       [:set-active-document-title! route-param])}))
 
 (rf/reg-event-fx
   :show-platform
-  (fn [_ route-param]
+  (fn [_ [_ route-param]]
     {:dispatch-n (list [:set-active-view :filtered-activities-view "platform"]
-                       [:filter-activities-by-search-term (second route-param)])}))
+                       [:filter-activities-by-search-term route-param]
+                       [:set-active-document-title! route-param])}))
 
 (rf/reg-event-fx
-  :show-branch
-  (fn [_ route-param]
+  :show-skill
+  (fn [_ [_ route-param]]
     {:dispatch-n (list [:set-active-view :filtered-activities-view "branch"]
-                       [:filter-activities-by-search-term (second route-param)])}))
+                       [:filter-activities-by-search-term route-param]
+                       [:set-active-document-title! route-param])}))
+
+(rf/reg-event-fx
+  :show-activity
+  (fn [_ [_ route-param]]
+    {:dispatch-n (list [:set-active-view :activity-view]
+                       [:set-activity-in-view route-param])}))
```

Comment:
makes sense! i'd like to merge this branch and open a new one for refactoring `:set-active-document-title!` and `:filter-activities-by-search-term`. could i get a review for merging @tylerperkins @dviramontes ?